### ISSUE TITLE
Commit db3d30dcad breaks automatic tunnelling.

### ISSUE THREAD

**Issue by Parakleta** [TurboVNC/turbovnc#321]
In db3d30dcad you changed the tunnelling handling with the session manager to not change the provided option, but rather to use a local variable calculation.  This then breaks behaviour in the `fillCmdPattern()` method is `Tunnel.java` since it is still looking at the manually set option rather than using the session manager behaviour.  The documentation in `Params.java` says that the tunnel options if "effectively set" when the session manager is being used, but this is no longer entirely true.

https://github.com/TurboVNC/turbovnc/blob/f2b115f207739a1dd2bedcb5a246179e943ee1d6/java/com/turbovnc/vncviewer/Tunnel.java#L284-L285

https://github.com/TurboVNC/turbovnc/blob/f2b115f207739a1dd2bedcb5a246179e943ee1d6/java/com/turbovnc/vncviewer/Tunnel.java#L308-L309

**Comment by dcommander** [comment]
The code prior to that commit did not change `opts.tunnel` in **Tunnel.java**, so I don't understand what the issue is.  Can you provide an example of the breakage?

**Comment by Parakleta** [comment]
In `CConn.java` the `opts.tunnel` option is no longer forced true when the session manager is used.  This is worked around in the `createTunnel()` method in `Tunnel.java` by the local `tunnel` variable, but this variable is not propagated into the `fillCmdPattern()` method in the same file.

The side effect of this is that the two referenced lines now operate differently to before that change.  That is, the `%H` replacement no longer contains the username, and the `%G` token must be provided in the tunnel command or else an error is thrown.

**Comment by Parakleta** [comment]
This is not necessarily a problem, except that it is a breaking change that isn't documented, and in fact the relevant documentation for the tunnel option suggests that it should be automatically set by the session manager.  See:

https://github.com/TurboVNC/turbovnc/blob/473e569bcde180f796a9224288c7033a87be1c84/java/com/turbovnc/rfb/Params.java#L830-L835

and

https://github.com/TurboVNC/turbovnc/blob/473e569bcde180f796a9224288c7033a87be1c84/java/com/turbovnc/rfb/Params.java#L874-L884

**Comment by dcommander** [comment]
I don't understand your point.  External SSH clients are not yet supported with the Session Manager.  (See #148.) The modes that should work are:
- Internal SSH client with Session Manager
- Internal SSH client with explicit SSH tunneling (i.e. `Tunnel` parameter set)
- External SSH client with explicit SSH tunneling

If one of those modes does not work, then please provide exact steps to reproduce the failure.

**Comment by dcommander** [comment]
Ah.  I think I see the issue.  When `VNC_TUNNEL_CMD` is set, the `createTunnel()` function assumes that an external SSH client will be used, which is the wrong assumption when the session manager is active.  Stand by.

**Comment by dcommander** [comment]
Yes, it is effectively the same as if you had opened a separate SSH session in another window.  Port forwarding is only enabled for the viewer SSH session, not the Session Manager SSH session.  (When using the built-in SSH client, the two sessions are the same.)

I will go ahead and fix the issue.  Just understand that using `ExtSSH` with the Session Manager isn't officially supported yet, and without public key authentication, that configuration will require that you authenticate twice.  (Officially supporting `ExtSSH` with the Session Manager will require using the OpenSSH ControlMaster option to enable reuse of the SSH session when using an external SSH client.)
